packages/headless/src/api/knowledge/createAnswerAPISlice.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  EventSourceMessage,
+  fetchEventSource,
+} from '@microsoft/fetch-event-source';
+import {ConfigurationSection} from '../../state/state-sections';
+import {createAbortController} from '../../utils/abort-controller-polyfill';
+import {baseKnowledgeAPISlice} from './baseKnowledgeAPISlice';
+
+type StateNeededByKnowledgeAPI = ConfigurationSection;
+
+interface ErrorPayload {
```

Comment:
There are already error types defined in `platform-client.ts` and related: Need to unify those and not create more error interface.

Idem for `FatalError`
I know they come from `generated-answer-client`, but... I would like if we could think about unifying those a bit. There is no reasons to have as many different error class and types.

Reply:
Fair enough :pray:

Reply:
Looking a this a bit more closely. I am not sure what to do.
First of all, I did not question to much how the error management for the actual stream was made. I rather wanted to keep is as close at its actually handling. Maybe this is the right decision, maybe not, but all in all, this bring me to the following point.

As you certainly noticed, the `ErrorPayload` interface and the `FatalError` class are used to make trigger different behaviors according to the error code coming from the API. There might be a better pattern to respect this behavior, but I cannot see how to keep the current logic without them. And I do not see anything in `platform-client.ts` that I can use to handle a stream. 
I do understand the unification concern. But what would you suggest to avoid declaring those here ?